network/p2p/middleware.go

```diff
@@ -166,28 +182,16 @@ func (m *Middleware) Start(ov network.Overlay) error {
 		pubsub.WithMaxMessageSize(m.maxPubSubMsgSize),
 	}
 
+	// creates libp2p host and node
 	nodeAddress := NodeAddress{Name: m.me.String(), IP: m.host, Port: m.port}
-
-	libp2pKey, err := privKey(m.key)
-	if err != nil {
-		return fmt.Errorf("failed to translate Flow key to Libp2p key: %w", err)
-	}
-
-	// start the libp2p node
-	err = m.libP2PNode.Start(m.ctx,
-		nodeAddress,
-		m.log, libp2pKey,
-		m.handleIncomingStream,
-		m.rootBlockID,
-		true,
-		nodeAddrsWhiteList,
-		m.metrics,
-		psOptions...)
+	libP2PNode, err := m.libP2PNodeFactory(m, nodeAddress, nodeAddrsWhiteList, psOptions...)
```

Comment:
We move the key translation functionality to where it really belongs and create the libp2p node based on the injected factory.